api/python/cellxgene_census/src/cellxgene_census/experimental/ml/pytorch.py

```diff
@@ -36,6 +36,54 @@
 The Tensors are rank 1 if ``batch_size`` is 1, otherwise the Tensors are rank 2."""
 
 
+class Encoder(abc.ABC):
+    """Base class for obs encoders.
+    To define a custom encoder, two methods must be implemented:
+
+    - ``register``: defines how the encoder will be fitted to the data.
+    - ``transform``: defines how the encoder will be applied to the data
+      in order to create an obs_tensor.
+
+    See the implementation of ``DefaultEncoder`` for an example.
+    """
+
+    @abc.abstractmethod
+    def register(self, obs: pd.DataFrame) -> None:
+        """Register the encoder with obs."""
+        pass
+
+    @abc.abstractmethod
+    def transform(self, df: pd.DataFrame) -> pd.DataFrame:
+        """Transform the obs DataFrame into a DataFrame of encoded values."""
+        pass
```

Comment:
⚠️ should also have an `inverse_transform` method, as LabelEncoder provided.. See example usage in `pytorch.ipynb`.  This was a hidden requirement...

Reply:
Right, I saw that but I forgot! Thanks for the catch!

Reply:
Added.